packages/core/src/state-management/state-manager.ts

```diff
@@ -340,14 +342,69 @@ export class StateManager {
     return this._processAnchorResponse(state$, anchorStatus$)
   }
 
-  private async _saveAnchorRequestForState(state$: RunningState): Promise<void> {
+  private async _buildAnchorRequestCARFile(streamId: StreamID, tip: CID): Promise<CAR> {
+    const carFactory = new CARFactory()
+    carFactory.codecs.add(DAG_JOSE)
```

Comment:
Not super important but why do you create a new factory every time you call this function? You could just have a global factory object.